### PR TITLE
add copyto, resize, histogram

### PR DESCRIPTION
**Pull request by ev-br**
on top of gh-105

### CODE REVIEW

File: torch_np/_funcs_impl.py [hunk 1]
```diff
@@ -987,6 +996,65 @@ def tile(A: ArrayLike, reps):
...
+        return torch.zeros(new_shape, dtype=a.dtype)
+
+    repeats = -(-new_size // a.numel())  # ceil division
+    a = concatenate((a,) * repeats)[:new_size]
+
+    return reshape(a, new_shape)
+
+
+def _ndarray_resize(a: ArrayLike, new_shape, refcheck=False):
+    # implementation of ndarray.resize.
```
Comment (lezcano) [inline torch_np/_funcs_impl.py:1027]: my my...

**Review by lezcano — APPROVED** [review]

File: torch_np/_funcs_impl.py (hunk 1 above)
Comment (lezcano) [reply torch_np/_funcs_impl.py:1027]: (I wanted to reference the next line)

**Comment by ev-br** [comment]
Am folding it into gh-105 to bring `copyto` to address review comments.

File: torch_np/_funcs_impl.py [hunk 2]
```diff
@@ -36,6 +38,13 @@ def copy(a: ArrayLike, order="K", subok: SubokLike = False):
     return a.clone()
 
 
+def copyto(dst: NDArray, src: ArrayLike, casting="same_kind", where=NoValue):
+    if where is not NoValue:
+        raise NotImplementedError
+    (src,) = _util.typecast_tensors((src,), dst.tensor.dtype, casting=casting)
+    dst.tensor.copy_(src)
```
Comment (lezcano) [inline torch_np/_funcs_impl.py:45]: Actually, this should be `dst.copy_(src)`?

Comment (ev-br) [reply torch_np/_funcs_impl.py:45]: Not really. `dst` is annotated as NDArray and those do not decay to torch.Tensors. src is an ArrayLike, so it can be a list to be normalized to a tensor. But `dst` cannot be a generic ArrayLike and must be an ndarray.

Comment (lezcano) [reply torch_np/_funcs_impl.py:45]: Sure, it must be an ndarray, but at this point, after the normalizer, we should have just PyTorch-land objects. In particular, we should never get `tnp.ndarray`s of any kind, just Tensors.

Comment (ev-br) [reply torch_np/_funcs_impl.py:45]: `dst: NDArray` annotation only checks that its argument is an ndarray exactly and does not convert it to a Tensor:

https://github.com/Quansight-Labs/numpy_pytorch_interop/blob/main/torch_np/_normalizations.py#L62

Comment (lezcano) [reply torch_np/_funcs_impl.py:45]: Yeah. My point then is that `NDArray` should convert it to a `Tensor`. Why would it not do so?

Comment (ev-br) [reply torch_np/_funcs_impl.py:45]: Why would it. It's a rare annotation for when need to only accept an ndarray not a general array_like. And then we know it has a `.tensor` attribute. It's a rare annotation, currently used here and for the out= arrays, which also need to be exactly arrays.

Comment (lezcano) [reply torch_np/_funcs_impl.py:45]: It currently breaks the invariant that every input to these functions is something that lives in PyTorch-land. I would expect this to be a more constrained version of `ArrayLike`, but currently it has other semantics, which is a bit odd.

Comment (ev-br) [reply torch_np/_funcs_impl.py:45]: Okay. So does it hold gh-105?

Comment (lezcano) [reply torch_np/_funcs_impl.py:45]: It can be done afterwards as a small ENH.

Comment (ev-br) [reply torch_np/_funcs_impl.py:45]: https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/108
